docs/Technical-Documentation/tech-memos/multi-select-fiters/multi-select-filters.md

```diff
@@ -0,0 +1,171 @@
+              // All multi select filters are required to set the `key` and `value` fields on the option element to the
+              // individual options to be able to build the correct query string.
+              let selected = ''
+              for (const option of filter.options) {
+                if (option.selected) {
+                  selected = selected.concat(option.value, '%2C')
+                }
+              }
+              selected = selected.substring(0, selected.lastIndexOf('%2C'))
+              if (selected !== '') {
+                query = query.concat(conjunction, filter.options[0].getAttribute('key'), '=', selected)
+              }
+            }
+          }
+          window.location = query
+        };
+      }
+    });
+  })(django.jQuery);
+}
+```
+
+### TDP Updates
+Currently, TDP implements a custom multi-select filter, with a custom template. This filter is complex and relies on a custom "filter" button to apply it's selection which is a very incohesive with the `Apply Filters` button that 508 introduces. To remedy this, the current and future multi-select/custom filters implemented in TDP need to give 508 control of constructing the appropriate query string by way of supplying 508 with the appropriate key-value pairs needed. In doing so, we can also simplify and generalize the current multi-select filter available in TDP.
+
+TDP currently utilizes three classes to implement field based multi-select filtering. This can be simplified to the single class below when we let 508 manage query string building. There are three main features to note in this class. The first is the custom template used to create a multi-select drop down filter: [multiselectdropdownfilter.html](multiselectdropdownfilter.html). The second is the unique query string parameters that are defined in the `choices` method of the `FieldListMultiSelectFilter` class: `key` and `value`. These are the parameters that Django will populate into the aforementioned template and that 508 will parse to build the appropriate query string. The third and final feature to note is how the `FieldListMultiSelectFilter` overrides and adds new parameters in the constructor before calling `super()`. These additions and overrides help convert the parent class `AllValuesFieldListFilter` from a single-select to a multi-select filter. These three features allow TDP to support multi-select filtering based on Django model fields. However, this MVP implementation also provides a path forward for Django model property based multi-select filtering (e.g. the `fiscal_period` property on the `DataFile` model). Leveraging the aforementioned template and building a class which sub-classes the Django `SimpleListFiter` class we will have the ability to provide general multi-select filtering for Django model properties by implementing the correct `queryset`, `choices`, and `lookups` methods.
+
+```python
+class FieldListMultiSelectFilter(admin.AllValuesFieldListFilter):
```

Comment:
Not sure if adding the whole logic is useful. What's the plan? to delete this memo later when this is implemented? I suggest we define the direction and path forward

Reply:
Right, I think thats what it does. It provides the suggested implementation for our current multi select filtering needs and points out how we can generalize it further. I will change the language a bit to make it seem like the class has to be used since it was an example implementation.

Reply:
Let me know what you think of the changes. Happy to keep iterating